test/tools/spec-runner/index.js

```diff
@@ -132,12 +132,17 @@ function generateTopologyTests(testSuites, testContext, filter) {
 
       const allRequirements = runOn.map(legacyRunOnToRunOnRequirement);
 
-      let shouldRun = true;
+      let someRequirementMet = allRequirements.length ? false : true;
       for (const requirement of allRequirements) {
-        shouldRun =
-          shouldRun && (await topologySatisfies(this.currentTest.ctx, requirement, utilClient));
+        const met = await topologySatisfies(this.currentTest.ctx, requirement, utilClient);
+        if (met) {
+          someRequirementMet = true;
+          break;
+        }
       }
 
+      let shouldRun = someRequirementMet;
+
```

Comment:
same comment as in `unified-spec-runner/runner` - abstracting this to a function would make this a lot easier to understand.  If we moved the loop into a function, we could write it like this

```typescript
const someRequirementMet = allRequirements.length > 0 && await isAnyRequirementMet(....) 
```